common/detector.py

```diff
@@ -13,17 +13,25 @@ def __init__(self, regular_mbox):
         self.inbox = regular_mbox
 
     @abc.abstractmethod
+    def update_sender_profile(self, email):
+        """Updates sender to profile map with a single email.
+
+        Keyword arguments:
+        email -- the email message to add to the sender profile
+        """
+        return
+
     def create_sender_profile(self, num_samples):
         """Creates sender to profile map.
 
         Keyword arguments:
         num_samples -- number of samples to train sender profile on.
 
         Sets self.sender_profile to a dictionary mapping senders to profiles.
-
-        Returns self.sender_profile.
         """
-        return
+        for i in range(num_samples):
```

Comment:
Do you think we should initialize `self.sender_profile = {}` here before the for loop? Right now, we handle that in the constructor for each detector, but that could lead to a scenario where someone calls `create_sender_profile` twice and the 2nd invocation does not start from an "empty" sender profile. On the other hand, many of our detectors require additional data structures anyway for the sender profile, so another alternative would be to leave this as an abstract method and have each individual detector move the sender profile initialization to this method. Thoughts?